aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/NativeClient.java

```diff
@@ -13,15 +13,19 @@
  */
 class NativeClient {
     private static final String nativeLibPath = "/tmp/.aws-lambda-runtime-interface-client";
+    private static final String architecturePathSuffix = "/" + getArchIdentifier();
+    private static final String supported_arm_architectures = "^(aarch64)$";
+    private static final String supported_x86_architectures = "^(amd64)$";
```

Comment:
This is still very restrictive and may not cater for all supported architectures - please change to the one proposed in my previously linked commit:

```java
        if (arch.matches("^(x8664|amd64|ia32e|em64t|x64|x86_64)$")) {
            return "x86_64";
        } else if (arch.startsWith("arm64") || arch.startsWith("aarch64")) {
            return "arm64";
        }
```

Also please include a comment describing where these patterns come from as suggested:

```java
// Implementation based on AWS CRT (ref. https://github.com/awslabs/aws-crt-java/blob/0e9c3db8b07258b57c2503cfc47c787ccef10670/src/main/java/software/amazon/awssdk/crt/CRT.java#L106-L134)
```

Reply:
I added the other architectures, to be less restrictive. But, I'm still using the regexp matches, so that we can still use a variable, holding all supported architectures. Hope this make sense to you.

Reply:
That's cool, thanks @muellerc - could you please also add the comment I mentioned above with a link to the source of the regex/logic for future reference?